packages/graphback-datasync/src/providers/DeltaDbDataProvider.ts

```diff
@@ -0,0 +1,178 @@
+      .forEach((f: FieldTransform) => {
+        data[f.fieldName] = f.transform(f.fieldName);
+      });
+
+    const objectId = new ObjectId(idField.value);
+    const result = await this.db.collection(this.collectionName).findOneAndUpdate({ _id: objectId }, { $set: data }, { returnOriginal: false });
+    if (result.ok) {
+      
+      this.deltaProvider.insertDiff(result.value).catch((e: any) => {
+        console.error(`Error: Couldn't insert into delta table: ${e}`);
+      })
+
+      return this.mapFields(result.value)
+      
+    }
+    throw new NoDataError(`Cannot update ${this.collectionName}`);
+
+
+  }
+
+  public async delete(data: any, context: GraphbackContext): Promise<Type> {
+    const conflict = await this.checkForConflicts(data, context);
+    if (conflict !== undefined) {
+      throw new ConflictError(conflict);
+    }
+
+    const { idField } = getDatabaseArguments(this.tableMap, data);
+    data = {};
+
+    this.fieldTransformMap[TransformType.UPDATE]
+      .forEach((f: FieldTransform) => {
+        data[f.fieldName] = f.transform();
+      });
+
+    data._deleted = true;
+    const objectId = new ObjectId(idField.value);
+    const projection = this.buildProjectionOption(context);
+    const result = await this.db.collection(this.collectionName).findOneAndUpdate({ _id: objectId }, { $set: data }, {/*projection,*/ returnOriginal: false });
+    if (result.ok) {
+      this.deltaProvider.insertDiff(result.value).catch((e: any) => {
+        console.error(`Error: Couldn't insert into delta table: ${e}`);
+      })
+
+      return this.mapFields(result.value);
+    }
+
+    throw new NoDataError(`Could not delete from ${this.collectionName}`);
+  }
+
+  public async findOne(filter: any, context: GraphbackContext): Promise<Type> {
+    if (filter.id) {
+      filter = { _id: new ObjectId(filter.id) }
+    }
+    const projection = this.buildProjectionOption(context);
+    const query = this.db.collection(this.collectionName).findOne({
+      ...filter,
+      _deleted: { $ne: true}
+    }, { projection });
+    const data = await query;
+
+    if (data) {
+      return this.mapFields(data);
+    }
+    throw new NoDataError(`Cannot find a result for ${this.collectionName} with filter: ${JSON.stringify(filter)}`);
+  }
+
+  public async findBy(filter: any, context: GraphbackContext, page?: GraphbackPage, orderBy?: GraphbackOrderBy): Promise<Type[]> {
+    if (filter === undefined) {
+      filter = {};
+    }
+
+    filter._deleted = { ne: true };
+
+    return super.findBy(filter, context, page, orderBy);
+  }
+
+  public async count(filter: any): Promise<number> {
+    if (filter === undefined) {
+      filter = {};
+    }
+
+    filter._deleted = { ne: true};
+
+    return super.count(filter);
+  }
+
+  public sync(lastSync: string, context: GraphbackContext, filter?: any): Promise<Type[]> {
+
+    return this.deltaProvider.sync(parseInt(lastSync, 10)).then((res: any[]) => {
```

Comment:
So we went to the 1.5 - WE have separate delta table with duplicate of the data. (Not really delta table itself.)?

Reply:
It is open for discussion, really, I just wanted get some feedback, I haven't hacked around with it that much to make a decisive move *yet*

Reply:
yep. No worries. This works and it is very very well written. But we cannot get conflict information from it. Let's drop some short research to see how we can get that done

Reply:
> So we went to the 1.5 - WE have separate delta table with duplicate of the data. (Not really delta table itself.)?

This should be more of a changelog + auditing table. This article might be of help https://ludwigstuyck.wordpress.com/2013/04/04/history-tracking/